zebra-network/src/peer_set/candidate_set.rs

```diff
@@ -341,12 +345,24 @@ fn validate_addrs(
 
     // TODO:
     // We should eventually implement these checks in this function:
-    // - Zebra should stop believing far-future last_seen times from peers (#1871)
     // - Zebra should ignore peers that are older than 3 weeks (part of #1865)
     //   - Zebra should count back 3 weeks from the newest peer timestamp sent
     //     by the other peer, to compensate for clock skew
     // - Zebra should limit the number of addresses it uses from a single Addrs
     //   response (#1869)
 
-    addrs
+    let addrs: Vec<_> = addrs.into_iter().collect();
```

Comment:
Since this function will receive more code in the future, should this new code be in a separate function, so that all validation/filtering steps are listed here as function calls?

Reply:
Yes, I think we should add a separate function for each check - that will make refactoring easier.

However, we should do the proptests on `validate_peers`, to make sure the properties hold when all the changes are combined.